sonaranalyzer-dotnet/src/SonarAnalyzer.Common/SymbolicExecution/AbstractExplodedGraph.cs

```diff
@@ -146,19 +146,16 @@ public void Walk()
             OnExplorationEnded();
         }
 
-        internal void AddExplodedGraphCheck<T>(T check)
+        public T GetOrAddCheck<T>(Func<T> factory)

```

Comment:
Since the SE analyzer registers states that subscribe for check events, adding a check does not replace an existing one, but reuses it.